cloud/amazon/rds.py

```diff
@@ -586,9 +586,9 @@ def get_data(self):
         }
         if self.instance["VpcSecurityGroups"] is not None:
             d['vpc_security_groups'] = ','.join(x['VpcSecurityGroupId'] for x in self.instance['VpcSecurityGroups'])
-        if self.status == 'available':
-            d['endpoint'] = self.instance["Endpoint"]["Address"]
-            d['port'] = self.instance["Endpoint"]["Port"]
+        if "Endpoint" in self.instance and self.instance["Endpoint"] is not None:
+            d['endpoint'] = self.instance["Endpoint"].get('Address', None)
```

Comment:
If `self.instance['Endpoint']` is not `None`, how likely is it that `self.instance['Endpoint']` does not have an `Address` key?